pkg/util/ipvs/testing/fake.go

```diff
@@ -159,14 +169,9 @@ func (f *FakeIPVS) DeleteRealServer(serv *utilipvs.VirtualServer, dest *utilipvs
 		return fmt.Errorf("Failed to delete destination for service %v, service not found", key.String())
 	}
 	dests := f.Destinations[key]
-	var i int
-	for i = range dests {
-		if dests[i].Equal(dest) {
-			break
-		}
-	}
+	i := toRealServerNum(dests, dest)
```

Comment:
I would use `RealServerKey1 == RealServerKey2` to replace `dests[1].Equal(dests[2])`

Please reference what we did for VirtualServer and serviceKey